python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -280,7 +280,7 @@ def validate_authentication(cls, values):
 
         # ensure at least 1 method is provided
         check.invariant(
-            auths_set > 0,
+            auths_set > 0 or values.get("authenticator") is not None,
```

Comment:
I think for code readability it would be better to structure this validation like 
```python
@root_validator
def validate_authentication(cls, values):
     if values.get("authenticator", None) is None:
           # rest of the password auth goes here in the form of the original code 
```

i think that makes it more clear that we only do password config validation if there is no other authentication method specified

Reply:
I can add a comment here (or update the above one) to explain the logic, but I think it's still true that when `authenticator` exists, you want users to only supply zero or one of the password/keys.

In other words:
- If no authenticator, then exactly one of those
- If authenticator, then zero or one of those

Reply:
ok i understand now (just goes to show a comment will be useful!). i think adding a comment like you explained above will be good